domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs

```diff
@@ -6,39 +6,40 @@
 using Asp.Versioning;
 using DataContext.Models;
 using EnergyOrigin.ActivityLog.DataContext;
-using EnergyOrigin.TokenValidation.Utilities;
-using EnergyOrigin.TokenValidation.Values;
+using EnergyOrigin.TokenValidation.b2c;
 using FluentValidation;
 using FluentValidation.AspNetCore;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 
 namespace API.Transfer.Api.Controllers;
 
-[Authorize]
 [ApiController]
-[ApiVersion(ApiVersions.Version20240103)]
+[Authorize(Policy.Frontend)]
+[ApiVersion(ApiVersions.Version20240515)]
 [Route("api/transfer/transfer-agreement-proposals")]
 public class TransferAgreementProposalController(
     IValidator<CreateTransferAgreementProposal> createTransferAgreementProposalValidator,
-    IUnitOfWork unitOfWork)
+    IUnitOfWork unitOfWork,
+    IdentityDescriptor identityDescriptor,
+    AccessDescriptor accessDescriptor)
     : ControllerBase
 {
     /// <summary>
     /// Create TransferAgreementProposal
     /// </summary>
+    /// <param name="organizationId">Sender organization id</param>
     /// <param name="request">The request object containing the StartDate, EndDate and ReceiverTin needed for creating the Transfer Agreement.</param>
     /// <response code="201">Created</response>
     /// <response code="400">Bad request</response>
     /// <response code="409">There is already a Transfer Agreement with this company tin within the selected date range</response>
-    [Authorize(Policy = PolicyName.RequiresCompany)]
+    [HttpPost]
     [ProducesResponseType(typeof(TransferAgreementProposalResponse), 201)]
     [ProducesResponseType(typeof(void), 400)]
     [ProducesResponseType(typeof(void), 409)]
-    [HttpPost]
-    public async Task<ActionResult> CreateTransferAgreementProposal(CreateTransferAgreementProposal request)
+    public async Task<ActionResult> CreateTransferAgreementProposal([FromQuery] Guid organizationId, CreateTransferAgreementProposal request)
     {
-        var user = new UserDescriptor(User);
+        accessDescriptor.AssertAuthorizedToAccessOrganization(organizationId);
```

Comment:
Is this the way we should check if user has right and not if statement?

Reply:
I do not know. I did not change anything

Reply:
It's fine :D Just saw it's transfer that is special.